crates/block-producer/src/challenger.rs

```diff
@@ -852,3 +796,56 @@ fn extract_inputs(tx: &Transaction) -> HashSet<OutPoint> {
     let inputs = tx.raw().inputs().into_iter();
     inputs.map(|i| i.previous_output()).collect()
 }
+
+fn extract_challenge_target(cell: &ChallengeCell) -> Result<ChallengeTarget> {
+    let lock_args = {
+        let args: Bytes = cell.output.lock().args().unpack();
+        match ChallengeLockArgsReader::verify(&args.slice(32..), false) {
+            Ok(_) => ChallengeLockArgs::new_unchecked(args.slice(32..)),
+            Err(err) => return Err(anyhow!("invalid challenge lock args {}", err)),
+        }
+    };
+
+    Ok(lock_args.target())
+}
+
+fn validate_load_data_strategy_offchain(
+    mock_context: &OffChainMockContext,
+    global_state: GlobalState,
+    challenge_cell: &ChallengeCell,
+    context: VerifyContext,
+) -> Result<LoadDataStrategy> {
+    let challenge_target = extract_challenge_target(challenge_cell)?;
+
+    let verify = |strategy: LoadDataStrategy| -> Result<_> {
+        log::info!("validate cancel challenge with strategy {:?}", strategy);
+
+        let mock_output = mock_cancel_challenge_tx(
+            &mock_context.mock_rollup,
+            &mock_context.mock_poa,
+            global_state.clone(),
+            challenge_target.clone(),
+            context.clone(),
+            Some(strategy),
+        )?;
+
+        if mock_output.tx.as_slice().len() as u64 > MAX_CANCEL_TX_BYTES {
+            bail!("cancel tx exceeded");
+        }
+
+        verify_tx(
+            &mock_context.rollup_cell_deps,
+            TxWithContext::from(mock_output),
+            MAX_CANCEL_CYCLES,
+        )?;
+
+        Ok(strategy)
+    };
+
+    match verify(LoadDataStrategy::Witness) {
+        Err(err) => log::info!("cancel challenge by witness {}, try cell dep", err),
```

Comment:
Should we use log::warn here, since typical txs should be able to handled by the Witness load strategy?

Reply:
I'm not sure here. Ok, we should raise the alarm here.